spyder/plugins/editor/widgets/completion.py

```diff
@@ -24,48 +24,35 @@
 
 
-COMPLETION_ITEM_TEMPLATE = u"""
-<table width="{width}" height="{height}">
-    <tr>
-        <td valign="middle" style="color:{color}">
-            {completion}
-        </td>
-
-        <td valign="middle" align="right" float="right" style="color:{color}">
-            {type}
-        </td>
-        <td valign="top" align="right" float="right" width="{img_width}">
-            <img src="data:image/png;base64, {icon_provider}"
-                 height="{img_height}" width={img_width}/>
-        </td>
-    </tr>
-</table>
-"""
 DEFAULT_COMPLETION_ITEM_HEIGHT = 15
 DEFAULT_COMPLETION_ITEM_WIDTH = 250
 
 
 class CompletionWidget(QListWidget):
     """Completion list widget."""
-    ICONS_MAP = {CompletionItemKind.TEXT: 'text',
-                 CompletionItemKind.METHOD: 'method',
-                 CompletionItemKind.FUNCTION: 'function',
-                 CompletionItemKind.CONSTRUCTOR: 'constructor',
-                 CompletionItemKind.FIELD: 'field',
-                 CompletionItemKind.VARIABLE: 'variable',
-                 CompletionItemKind.CLASS: 'class',
-                 CompletionItemKind.INTERFACE: 'interface',
-                 CompletionItemKind.MODULE: 'module',
-                 CompletionItemKind.PROPERTY: 'property',
-                 CompletionItemKind.UNIT: 'unit',
-                 CompletionItemKind.VALUE: 'value',
-                 CompletionItemKind.ENUM: 'enum',
-                 CompletionItemKind.KEYWORD: 'keyword',
-                 CompletionItemKind.SNIPPET: 'snippet',
-                 CompletionItemKind.COLOR: 'color',
-                 CompletionItemKind.FILE: 'filenew',
-                 CompletionItemKind.REFERENCE: 'reference',
-                 }
+    ITEM_TYPE_MAP = {CompletionItemKind.TEXT: 'text',
+                     CompletionItemKind.METHOD: 'method',
+                     CompletionItemKind.FUNCTION: 'function',
+                     CompletionItemKind.CONSTRUCTOR: 'constructor',
+                     CompletionItemKind.FIELD: 'field',
+                     CompletionItemKind.VARIABLE: 'variable',
+                     CompletionItemKind.CLASS: 'class',
+                     CompletionItemKind.INTERFACE: 'interface',
+                     CompletionItemKind.MODULE: 'module',
+                     CompletionItemKind.PROPERTY: 'property',
+                     CompletionItemKind.UNIT: 'unit',
+                     CompletionItemKind.VALUE: 'value',
+                     CompletionItemKind.ENUM: 'enum',
+                     CompletionItemKind.KEYWORD: 'keyword',
+                     CompletionItemKind.SNIPPET: 'snippet',
+                     CompletionItemKind.COLOR: 'color',
+                     CompletionItemKind.FILE: 'filenew',
+                     CompletionItemKind.REFERENCE: 'reference',
+                     }
+
+    ICON_MAP = {typ: ima.icon(typ) for typ in set(ITEM_TYPE_MAP.values())}
+    ICON_MAP['no_match'] = ima.icon('no_match')
```

Comment:
@metalogical @ccordoba12 @goanpeca checking through SSH why the tests on CircleCI are failing, seems like the above lines are some how triggering the error. I tested by changing the call from `ima.icon()` to an empty `QIcon` and then I'm able to run `reporterror.py` (I had to change both lines):

```python
    ICON_MAP = {typ: QIcon() for typ in set(ITEM_TYPE_MAP.values())}
    ICON_MAP['no_match'] = QIcon()
```

Reply:
instead of eagerly precaching these, I can just use a lazy cache.

Reply:
We can simply skip this file from CircleCI because there are no tests in the file itself (CircleCI runs very simple tests that we have at the end of each file).

I'll push a commit here to take care of that.

Reply:
I understood this incorrectly, sorry. It's making to fail all files that use QtAwesome.

So @metalogical, please implement your suggestion above:

> instead of eagerly precaching these, I can just use a lazy cache.

to see if that fixes the problems seen in CircleCI.